tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -88,6 +91,103 @@ def tearDown(self):
 
 
 class EndToEndShadowIndexingTest(EndToEndShadowIndexingBase):
+    def _all_uploads_done(self):
+        topic_description = self.rpk.describe_topic(self.topic)
+        partition = next(topic_description)
+
+        hwm = partition.high_watermark
+
+        manifest = None
+        try:
+            bucket = BucketView(self.redpanda)
+            manifest = bucket.manifest_for_ntp(self.topic, partition.id)
+        except Exception as e:
+            self.logger.info(
+                f"Exception thrown while retrieving the manifest: {e}")
+            return False
+
+        top_segment = max(manifest['segments'].values(),
+                          key=lambda seg: seg['base_offset'])
+        uploaded_raft_offset = top_segment['committed_offset']
+        uploaded_kafka_offset = uploaded_raft_offset - top_segment[
+            'delta_offset_end']
+        self.logger.info(
+            f"Remote HWM {uploaded_kafka_offset} (raft {uploaded_raft_offset}), local hwm {hwm}"
+        )
+
+        # -1 because uploaded offset is inclusive, hwm is exclusive
+        if uploaded_kafka_offset < (hwm - 1):
+            return False
+
+        return True
+
+    @cluster(num_nodes=7)
+    @matrix(cloud_storage_type=get_cloud_storage_type())
+    def test_reset(self, cloud_storage_type):
+        brokers = self.redpanda.started_nodes()
+
+        msg_count_before_reset = 50 * (self.segment_size // 2056)
+        producer = KgoVerifierProducer(self.test_context,
+                                       self.redpanda,
+                                       self.topic,
+                                       msg_size=2056,
+                                       msg_count=50 *
+                                       (self.segment_size // 2056),
+                                       debug_logs=True,
+                                       trace_logs=True)
+
+        producer.start()
+        producer.wait(timeout_sec=60)
+
+        wait_until(lambda: self._all_uploads_done() == True,
+                   timeout_sec=60,
+                   backoff_sec=5)
+
+        s3_snapshot = BucketView(self.redpanda, topics=self.topics)
+        manifest = s3_snapshot.manifest_for_ntp(self.topic, 0)
+
+        self.rpk.alter_topic_config(self.topic, 'redpanda.remote.write',
+                                    'false')
+        time.sleep(1)
+
+        json_man = json.dumps(manifest)
+        self.logger.info(f"unsafe_manifest={json_man}")
+        self.redpanda._admin.unsafe_reset_cloud_metadata(
+            self.topic, 0, manifest)
```

Comment:
As written, this could be a no-op and the test would still pass -- it would be better to make some edits to the manifest and confirm they're reflected in the result.

Reply:
Right. I had tried with edited manifests locally, but it's a good point. Updated the tests to shave some segments off.